presto-main/src/test/java/com/facebook/presto/type/TestTimestamp.java

```diff
@@ -237,6 +243,49 @@ public void testCastFromSlice()
     }
 
     @Test
+    public void testCastFromSliceFailsOnTimestampWithTimeZone()
+    {
+        if (connectorSession().isLegacyTimestamp()) {
+            // we expect those not to fail in legacy mode
+            return;
```

Comment:
If you don't mind I will just not care. Those test have variants for both scenarios and I don't want to waste time for making tests more perfect in a transition phase.